cmd/create.go

```diff
@@ -137,104 +202,341 @@ type createConfig struct {
+		fmt.Fprintf(&b, "  %v\n", v)
+	}
+	return errors.New(b.String())
+}
+
+// Validate the current state of the config, returning any errors.
+// Note this is a deeper validation using a client already configured with a
+// preliminary config object from flags/config, such that the client instance
+// can be used to determine possible values for runtime, templates, etc.  a
+// pre-client validation should not be required, as the Client does its own
+// validation.
+func (c createConfig) Validate(client *fn.Client) (err error) {
+	// Client validates both language runtime and template exist, defaulting
+	// them if not (to 'node' and 'http', respectively).  However, if either of
+	// them are invalid, or the chosen combination does not exist, the error
+	// message is a rather terse one-liner.  This is suitable for libraries, but
+	// for a CLI it behooves us to be more verbose, including valid options for
+	// each.  So here, we check that the values entered (if any) are both valid
+	// and valid together.
+	if c.Runtime != "" && !isValidRuntime(client, c.Runtime) {
+		return newInvalidRuntimeError(client, c.Runtime)
+	}
+
+	if c.Template != "" && !isValidTemplate(client, c.Runtime, c.Template) {
+		return newInvalidTemplateError(client, c.Runtime, c.Template)
+	}
+
+	// TODO: refactor to be git-like with no name at time of creation, but rather
+	// with named deployment targets in a one-to-many configuration.
+	dirName, _ := deriveNameAndAbsolutePathFromPath(c.Path)
+	return utils.ValidateFunctionName(dirName)
+	// NOTE: perhaps additional validation would be of use here in the CLI, but
+	// the client libray itself is ultimately responsible for validating all input
+	// prior to exeuting any requests.
+}
+
+// prompt the user with value of config members, allowing for interactively
+// mutating the values. The provided clientFn is used to construct a transient
+// client for use during prompt autocompletion/suggestions (such as suggesting
+// valid templates)
+func (c createConfig) prompt(client *fn.Client) (createConfig, error) {
+	var qs []*survey.Question
+
+	runtimes, err := client.Runtimes()
+	if err != nil {
+		return createConfig{}, err
+	}
+
+	// First ask for path...
+	qs = []*survey.Question{
 		{
 			Name: "path",
 			Prompt: &survey.Input{
-				Message: "Project path:",
+				Message: "Function Path:",
 				Default: c.Path,
 			},
 			Validate: func(val interface{}) error {
 				derivedName, _ := deriveNameAndAbsolutePathFromPath(val.(string))
 				return utils.ValidateFunctionName(derivedName)
 			},
-		},
-		{
-			Name: "runtime",
+			Transform: func(ans interface{}) interface{} {
+				_, absolutePath := deriveNameAndAbsolutePathFromPath(ans.(string))
+				return absolutePath
+			},
+		}, {
+			Name: "language", // users refer to this as Language Runtime (language for short)
 			Prompt: &survey.Select{
-				Message: "Runtime:",
-				Options: buildpacks.RuntimesList(),
+				Message: "Language Runtime:",
+				Options: runtimes,
```

Comment:
All references in code use the more accurate "runtime".  Help text and user documentation using the full name "Language Runtime" and associated flags [--language, -l]`